django/thunderstore/api/cyberstorm/tests/test_packages.py

```diff
@@ -0,0 +1,562 @@
+from typing import Dict, List, Union
+
+import pytest
+from django.http import HttpRequest
+from django.urls import reverse
+from rest_framework.exceptions import ParseError
+from rest_framework.test import APIClient
+
+from thunderstore.api.cyberstorm.views.package_preview import PackagePreviewAPIView
+from thunderstore.cache.enums import CacheBustCondition
+from thunderstore.cache.tasks import invalidate_cache
+from thunderstore.community.consts import PackageListingReviewStatus
+from thunderstore.community.factories import CommunitySiteFactory, PackageListingFactory
+from thunderstore.community.models import PackageCategory, PackageListingSection
+from thunderstore.community.models.package_listing import PackageListing
+from thunderstore.repository.factories import (
+    PackageFactory,
+    PackageRatingFactory,
+    PackageVersionFactory,
+    TeamFactory,
+    TeamMemberFactory,
+)
+
+PARAMS_FIXTURE = {
+    "community_identifier": "test",
+    "namespace_identifier": "test",
+    "team_identifier": "test",
+    "user_identifier": "test",
+    "package_identifier": "test",
+    "deprecated": True,
+    "nsfw": True,
+    "included_categories": ["a", "b"],
+    "excluded_categories": ["a", "b"],
+    "section": "test",
+    "q": "test",
+    "ordering": "last-updated",
+    "page": 1,
+    "page_size": 1,
+}
+
+
+@pytest.fixture(scope="module", autouse=True)
+def clear_pagination_cache():
+    invalidate_cache(CacheBustCondition.any_package_updated)
+
+
+@pytest.mark.django_db
+def test_PackagePreviewAPIView_get() -> None:
+    view = PackagePreviewAPIView()
+    request = HttpRequest()
+    setattr(request, "query_params", "bad")
+    assert view.get(request).status_code == 400
+    setattr(request, "query_params", {"page_size": 1})
+    assert view.get(request).status_code == 200
+
+
+@pytest.mark.django_db
+def test_PackagePreviewAPIView_get_full_cache_vary() -> None:
+    view = PackagePreviewAPIView()
+    assert (
+        view.get_full_cache_vary(PARAMS_FIXTURE)
+        == "package_cache_key.test.test.test.test.test.True.True.['a', 'b'].['a', 'b'].test.test.last-updated.1.1"
```

Comment:
Since same values (e.g. "test") are repeated in the fixture, there's no way of knowing that the returned string contains the values in correct, stable order (or that they're all included and not repeated, for that matter).